docs/tutorial.rst

```diff
@@ -1078,46 +1083,78 @@ of at least 0.8.
         )
     )
     mut_id = len(contig.mutation_types) - 1
-    # We will place our mutation in the middle of the contig.
+
+Next, we will set up the "extended events" which will modify the demography.
+The first extended event is placing of the selected mutation,
+which will occur in a random individual from the first population (id 0),
+in the middle of the contig, 1000 generations ago.
+We specify ``save=True`` to ``stdpopsim.ext.DrawMutation``
+so that the simulation can restart from that point if the mutation is lost.
+
+.. code-block:: python
+
     coordinate = round(contig.recombination_map.sequence_length / 2)
-    # Time mutation will appear.
     T_mut = 1000
     extended_events = [
-        # Draw the mutation
         stdpopsim.ext.DrawMutation(
             time=T_mut,
             mutation_type_id=mut_id,
             population_id=0,
             coordinate=coordinate,
-            # Save state before the mutation is introduced.
-            # This allows the simulation to restart from here if the mutation is lost.
             save=True,
-        ),
-        # Condition on not loosing the mutation from the generation the mutation
-        # is added until the end. Note that our conditioning must start one
-        # generation after the mutation is placed. You cannot simply specify
-        # T_mut+1 if rescaling is present, otherwise the conditioning would start
-        # at the same generation when the mutation is placed.
-        stdpopsim.ext.ConditionOnAlleleFrequency(
-            start_time=stdpopsim.ext.GenerationAfter(T_mut),
-            end_time=0,
-            mutation_type_id=mut_id,
-            population_id=0,
-            op=">",
-            allele_frequency=0.0,
-        ),
-        # At the end of the simulation, we want our swept mutation to have
-        # reached at least 0.5 frequency.
-        stdpopsim.ext.ConditionOnAlleleFrequency(
-            start_time=0,
-            end_time=0,
-            mutation_type_id=mut_id,
-            population_id=0,
-            op=">=",
-            allele_frequency=0.8,
-        ),
+        )
     ]
-    # Simulate.
+
+Next,we condition on the mutation not being lost.
+Since in the next step we condition on the mutation being at 80% frequency at
+the end, this is redundant, but it allows the simulation to immediately restart
+from any generation in which the mutation is lost, rather than waiting until the end.
+Note that this conditioning must start one
+generation after the mutation is placed, for which we use
+``stdpopsim.ext.GenerationAfter(T_mut)``.
+We cannot simply specify ``T_mut - 1`` if rescaling is present,
```

Comment:
it would be + 1 because this API has forward times

Reply:
good catch - I meant to ask. If it has forwards times, though, why are "start time" and "end time" equal to 0 for the final conditioning?

Reply:
The API here is confusing, sorry. Times have units of "generations ago", just like msprime. The start_time/end_time parameters for the `ChangeMutationFitness` and `ConditionOnAlleleFrequency` need `start_time => end_time`, which is opposite to what msprime uses (e.g in the demography debugger). I'm not sure why I did it that way.

There are various comments and docstring stubs in https://github.com/popsim-consortium/stdpopsim/blob/main/stdpopsim/ext/selection.py which explain some things.

Reply:
Ah-ha, thanks. ... so, should it have been `T-1` or `T+1` there?

Reply:
`T-1` looks right to me.

Reply:
gee, sorry for the confusion!